packages/superset-ui-legacy-preset-chart-nvd3/src/NVD3Vis.js

```diff
@@ -498,7 +498,7 @@ function nvd3Vis(element, props) {
     const isXAxisString = isVizTypes(['dist_bar', 'box_plot']);
     if (!isXAxisString && chart.xAxis && chart.xAxis.tickFormat) {
       chart.xAxis.tickFormat(xAxisFormatter);
-    } else {
+    } else if (chart.xAxis && chart.xAxis.tickFormat) {
```

Comment:
Could we combine the check with the condition above?

```
if (chart.xAxis && chart.xAxis.tickFormat) {
  if(isXAxisString) {
  } else {
  }
}
```

Reply:
Done!